test/apps/iodemo/io_demo.cc

```diff
@@ -346,7 +372,14 @@ class P2pDemoCommon : public UcxContext {
             m->op        = op;
             m->data_size = data_size;
             if (validate) {
-                m->hdr.chk_sum = ucs_crc16(&m->tr, sizeof(*m) - sizeof(m->hdr));
+                const uint8_t *end = reinterpret_cast<uint8_t*>(_buffer) +
+                                     _io_msg_size;
+                uint8_t *tail      = reinterpret_cast<uint8_t*>(m + 1);
+                while (tail < end) {
+                    *(tail++) = IoDemoRandom::rand<uint8_t>();
+                }
```

Comment:
unite common code with fill_data() ?

Reply:
`fill_data` is a private method of BufferIov which works directly with `Buffer`
`IoMessage::init` fills `void* _buffer` and has diff header
So, IMHO these funcs has more diff then common parts - just filling buffer with N bytes of random values.

Also `fill data` is implemented for better perf on larger buffers when io_msg is assumed not to be so big as data buffer

Reply:
so filling N rand values could be a common function

Reply:
fixed